PXDConverter/Program.cs

```diff
@@ -28,132 +46,549 @@ private static void Main(string[] args)
             else
-                ConvertOneFile(argPath, tmpPath);
+            {
+                if (argPath.EndsWith(".pxd"))
+                {
+                    var lib = new PXD32Library();
+                    lib.InitializeDll();
+                    ProcessSinglePXDFile(argPath, lib);
+                    lib.CloseDll();
+                }
+                else if (argPath.EndsWith("inf.bin"))
+                {
+                    var lib = new EjToolLibrary();
+                    ProcessMultiWithBinaryInf(argPath, new EjToolLibrary());
+                }
+                else if (argPath.EndsWith(".inf"))
+                {
+                    ProcessMultiWithTextInf(argPath, new EjToolLibrary());
+                }
+                else if (argPath.EndsWith(".scl"))
+                {
+                    SclToWav(argPath);
+                }
+                else
+                {
+                    Console.WriteLine("You should pass only *.PXD, *inf.bin and .inf files");
+                }
+            }
+
+            Directory.Delete(TmpPath, true);
 
-            MarshalService.CloseDll();
-            
             Console.WriteLine("Finished.");
         }
 
-        private static void ConvertOneFile(string argPath, string tmpPath)
+        private static void ProcessSinglePXDFile(string argPath, PXD32Library lib)
         {
-            MarshalService.ConvertWavToRawDataBuffer(argPath, tmpPath);
-            
-            Console.WriteLine("Converting raw data to wav format...");
+            // it can be already wav file
+            if (ASCIIEncoding.ASCII.GetString(ReadFileFromTo(argPath, 0, 4)) == "RIFF")
+            {
+                string dir = $"{Directory.GetCurrentDirectory()}\\converted_wav_files\\";
+                Directory.CreateDirectory(dir);
+                string filePath = $"{dir}{Path.GetFileNameWithoutExtension(argPath)}.wav";
+                if (!File.Exists(filePath))
+                {
+                    File.Copy(argPath, filePath);
+                }
+            }
+            else
+            {
+                Console.WriteLine($"Converting raw data file {argPath} to wav format...");
 
-            RawToWav(argPath, tmpPath);
+                string tmpPath = TmpPath.ToString() + "tmp.wav";
+
+                lib.Decompress(argPath, 0, 0, 0, 0, 0, tmpPath);
+
+                RawToWav(argPath, tmpPath);
+            }
         }
 
-        private static void ConvertMultipleFiles(string argPath, string tmpPath)
+        private record PXDHeader(string name, string package, string type, int samples, bool isWave);
+
+        private static PXDHeader readPXDFileHeader(string filepath, byte[] bytes)
         {
-            Console.WriteLine("Searching for all .pxd files in specified location...");
+            bool isWave = ASCIIEncoding.ASCII.GetString(bytes[0..4]) == "RIFF";
+            // rest is useless if it's wav file, but anyway
+            int p = 4;
```

Comment:
What's "p"? I assume it's "pointer", but I would simply give it a more appropriate name ;)
you use that name in several places.
what do you think?

Reply:
p like position, will rename this